### DISCUSSION TITLE
Different debug console behavior between zephyr 3.2 and 3.5

### DISCUSSION THREAD

**Discussion by MarsMSWu** [zephyrproject-rtos/zephyr#66526] [Q&A] [answered]
Hi, all

  My previous zephyr base is 3.2 and current zephyr base is 3.5.
  Both are them test in LPC55s69 EVB.
  My project folder and config file in base 3.2 is work fine, can output log and LED toggle by 1s timer.
  The debug log "SEC" also show by 1s timer...and button INTR also can detect by pressed 0 or 1.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/277ddbef-97d3-40f2-a452-174335a2ea75)
  But move my project folder and config file in base 3.5 is work abnormal.
  It seems like LED still can toggle by 1s timer but output log by 1s timer is not show and no button INTR event.
  No log can let me know is button INTR work or not but LED still can toggle by 1s timer...maybe the code is work
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/640807e5-2ebd-4d43-ae8b-605660b97e88)
  In my understanding, it should something difference between 3.2 and 3.5 for debug console control.

  But I also try hello world in base 3.5, it works fine and can always show debug log in while loop.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/d9743721-0c74-46a4-8201-df6865d10142)
  If so, it seems like 3.5 should be no impact for debug console.

  I had try to compare the config file, but not sure which one is important key config item for this behavior.
  I really have no idea, is any one can advice for me ?

  Thanks~

**Comment by MarsMSWu** [comment]
Hi all,

I have no idea for this behavior, so I try to not set CONFIG_LOG=y
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/60c76d48-6e5f-4798-8c7f-d8f3cd42a4f1)
After this try then the debug console can show more...even if 1s timer becomes abnormal
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/89f8f922-82c2-4af6-9dd9-f5ab867a4c69)

I don't know why... because CONFIG_LOG=y also set in base 3.2 and it works fine.

Thanks~

**Comment by DerekSnell** [comment]
Hi @MarsMSWu ,
It seems your app is dependent on the Logging subsystem.  And yes, when using Logging, then `CONFIG_LOGGING=y` should be selected.  You can read more at [Logging](https://docs.zephyrproject.org/latest/services/logging/index.html).  But it seems you found the key difference between the v3.2 build and your current build.

Best regards

**Reply by MarsMSWu** [reply]
Hi @DerekSnell ,
Do you mean you also agree "not set CONFIG_LOG=y" is correct solution for my case ?
BTW, CONFIG_LOGGING can't parse while compile...
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/a02c1c9e-ebe7-4ea6-99a0-3e08b6d0ce5c)

I try to compare the BIN file size, as below
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/10f51f54-a1b7-4972-9670-6e5bc4c39f29)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/fd01d2c7-03de-4f6b-a228-371a2bec70c5)
And
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/228a63c0-f315-48d9-93de-2e85e4990870)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/b5a3f4c2-71f3-4f91-a01d-3e7d3c9d139d)
The BINARY size difference is 23KB, is it correct ? Just "CONFIG_LOG=y" enabled or not...

Thanks.

**Reply by DerekSnell** [reply, marked as answer]
Hi @MarsMSWu ,
Sorry, CONFIG_LOGGING was an error.  As you found, the correct symbol is `CONFIG_LOG`.
> Do you mean you also agree "not set CONFIG_LOG=y" is correct solution for my case ?

I was confused by your previous description, and the details of this issue.  But based on your last response, and rereading this, it seems your v3.5 port only prints everything when `CONFIG_LOG=n`.  Is that correct?  I assume you are not using the Logging APIs then.  Are you printing to the console with `printk()`?

If your app is not using the Logging APIs and features, then you can disable that subsystem.  Many of the Zephyr drivers and subsystems use Logging, so you may find it helpful to enable for debugging.  If you are using `printk()` in your app, and it does not print with `CONFIG_LOG=y`, do you have the `CONFIG_LOG_PRINTK` Kconfig selected?  I also found this [helpful blog](https://blog.golioth.io/debugging-zephyr-for-beginners-printk-and-the-logging-subsystem/) from our friends at Golioth.

As for the memory used by the Logging subsystem: the [Logging](https://docs.zephyrproject.org/latest/services/logging/index.html) features are very configurable.  The more features enabled, the more memory it consumes.  If you want to include Logging with you app, you can try the different configurations to optimize the memory usage.

Best regards

**Reply by MarsMSWu** [reply]
Hi @DerekSnell ,
Yes, can output everything while CONFIG_LOG=n and we only use printk() for output.
Not use any LOG API function.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/82646c15-c42b-4e1c-be9d-e94b3d96ef29)
The CONFIG_LOG will impact my debug console output.
I try to keep enable CONFIG_LOG and add CONFIG_LOG_PRINTK=y then can't output everything.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/167df580-4bba-4401-8fd9-49343037f614)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/af3edb03-6f3e-4f82-aa7f-f81b25304a12)

I check my .config file of build folder, there is no CONFIG_LOG_PRINTK in the file.
Do you mean CONFIG_PRINTK ?
![image](https://github.com/zephyrproject-rtos/zephyr/assets/151608605/6782877e-7d27-4520-bcc9-11a9b16828f0)

Thanks.

**Reply by DerekSnell** [reply]
Hi @MarsMSWu ,
I am referring to [CONFIG_LOG_PRINTK](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LOG_PRINTK).
Best regards
